docker-compose.yml

```diff
@@ -12,70 +16,74 @@ services:
       MYSQL_RANDOM_ROOT_PASSWORD: "true"
     volumes:
       - ./migrations/base.sql:/docker-entrypoint-initdb.d/init.sql:ro
-      - db-data:/var/lib/mysql
+      - ./.db-data:/var/lib/mysql
     healthcheck:
       test: "/usr/bin/mysql --user=$$MYSQL_USER --password=$$MYSQL_PASSWORD --execute \"SHOW DATABASES;\""
       interval: 2s
       timeout: 20s
       retries: 10
-    # Uncomment to expose the port to the host machine
-    #ports:
-    #  - ${DB_PORT}:${DB_PORT}
 
   redis:
     image: bitnami/redis:latest
+    # ports:
+    #   - ${REDIS_PORT}:${REDIS_PORT}
     user: root
     volumes:
-      - redis-data:/bitnami/redis/data
+      - ./.redis-data:/bitnami/redis/data
     environment:
       - ALLOW_EMPTY_PASSWORD=yes
       - REDIS_PASSWORD=${REDIS_PASS}
-    # Uncomment to expose the port to the host machine
-    #ports:
-    #  - ${REDIS_PORT}:${REDIS_PORT}
+
+    ## application services
 
   bancho:
-    build:
-      context: .
-      dockerfile: ./docker/banchopy/Dockerfile
+    image: bancho:latest
+    ports:
+      - ${APP_PORT:-${SERVER_PORT}}:${APP_PORT:-${SERVER_PORT}}
     depends_on:
       mysql:
         condition: service_healthy
       redis:
         condition: service_started
     tty: true
+    init: true
```

Comment:
@osuAkatsuki/core-osu-services @osuAkatsuki/bancho-py-devs @tsunyoku this is a neat find

https://docs.docker.com/compose/compose-file/compose-file-v3/#init

Reply:
no more are the days of writing SIGTERM handlers